pkg/controller/credentialsrequest/credentialsrequest_controller_test.go

```diff
@@ -443,6 +444,97 @@ func TestCredentialsRequestReconcile(t *testing.T) {
 				return mockAWSClient
 			},
 		},
+		{
+			name: "no namespace condition",
+			existing: []runtime.Object{
+				testCredentialsRequest(t),
+			},
+			mockRootAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				return mockAWSClient
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cr := getCR(c)
+				assert.NotNil(t, cr.Status)
+				assert.NotNil(t, cr.Status.Conditions)
+				condition := utils.FindCredentialsRequestCondition(cr.Status.Conditions, minterv1.MissingTargetNamespace)
+				assert.NotNil(t, condition)
```

Comment:
We should probably be checking the state of the condition, possibly the reason as well. Would it be worth adding test.expectedConditions with a struct encompassing the type, reason, and state, and move this validate logic into the test loop itself so it's run for every test that specifies expected conditions.

Reply:
Missed one use of expectedConditions I think. Looks good otherwise. Lets squash and hope for a test pass.